### ISSUE TITLE
Unparsable section error when creating Materialized view in Snowflake dialect

### ISSUE THREAD

**Issue by arnoprenveille** [sqlfluff/sqlfluff#5417]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

Sqlfluff lint raises a PRS error when running the linter on a snowflake dialect SQL file containing a "CREATE MATERIALIZED VIEW" statement

### Expected Behaviour

Given a sql script named **script.sql** creating a materialized view, running the command `sqlfluff lint script.sql --dialect=snowflake` should not raise any error.

### Observed Behaviour

Given the file script.sql : 

```
CREATE OR REPLACE MATERIALIZED VIEW TEST_MATERIALIZED_VIEW (
    foo
) AS
SELECT
    foo
FROM
    some_table;`
```
Running the linter causes the following error : 
```
>sqlfluff lint script.sql --dialect snowflake
== [script.sql] FAIL                                                                                                                                                                                                       
L:   1 | P:  59 |  PRS | Line 1, Position 59: Found unparsable section: ' (\n   
                       | foo\n) AS\nSELECT\n    foo\nFROM\n    ...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Run `sqlfluff lint script.sql --dialect snowflake` on any script.sql file containing the creation of a materialized view.

### Dialect

The sqlfluff dialect is Snowflake

### Version

```
> sqlfluff --version
sqlfluff, version 2.0.3
```
(the error also occurs in version 2.3.5)

### Configuration

`.sqlfluff` : 
```
[sqlfluff]
dialect = snowflake
# Ignore rule that forbids using Keywords as identifiers
# Ignore rule that avoids having aliases during FROM or JOIN clauses
exclude_rules = L029, L031
fix_even_unparsable = False
templater = jinja
max_line_length = 120
large_file_skip_byte_limit = 30000


[sqlfluff:templater:jinja]
library_path = sqlfluff_libs

[sqlfluff:templater:jinja:context]
ds = 2022-01-01
ds_nodash = 20220101
next_ds = 2022-01-01
next_ds_nodash = 20220101
prev_ds = 2022-01-01
ts = 2022-01-01 10:00:00
ts_nodash = 20220101100000
data_interval_start = 2022-01-01

params = {
    'ARTICLE_SUBDOMAIN' : 'test_subdomain',
    'DWH_DB': 'test',
    'AGGREGATE_SCHEMA_NAME': 'test',
    'subscription_lookback_window': 1,
    'devices_lookback_window': 1,
    'bracket_open': '(',
    'bracket_close': ')',
    'PARTNERS': 'partners',
    'EXT_DB': 'test',
    'partnership_launch': '2022-01-01',
    'partner_label': 'partner',
    'CDM_SCHEMA_NAME': 'CDM',
    'product_ids': '(1, 2)',
    'registration_lookafter_window': '1',
    'registration_attribution_window': '1',
    'table_name': 'test',
    'anonymous_table_name' : 'anonymous',
    'id_column': 'test_column',
    'subscription_lookafter_window': '1',
    'subscription_attribution_window': '1',
    'subscription_attribution_window_in_days': '1',
    'publisher_ids' : '(1, 2, 3)',
    'columns_to_exclude': 'user_id',
    'anonymous_column_name': 'uuid',
    'start_date_id': 20220601,
    'end_date_id': 20220630,
    'minimum_value_to_generate': 0,
    'quarterly': True,
    'inclusion': (0,1),
    'upsell': (0,1),
    'label_partner': '0',
    'both': (0,1),
    'columns': ['id'],
    'columns_map': {'id': 'INTEGER'},
    'tables': ['test'],
    'event_name': 'email_sent'
    }

[sqlfluff:indentation]
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = False
template_blocks_indent = True

[sqlfluff:rules:capitalisation.keywords]
# Keywords must be uppercase
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
# Unquoted identifiers must be lowercase
extended_capitalisation_policy = lower

[sqlfluff:rules:references.consistent]
# References should be unqualified in statements with a single table
force_enable = True
single_table_references = unqualified

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:layout.select_targets]
wildcard_policy = multiple

[sqlfluff:rules:capitalisation.literals]
# Booleans and NULL must be uppercase
extended_capitalisation_policy = upper

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = both

[sqlfluff:rules:convention.terminator]
# SQL query must end with a semi-colon
require_final_semicolon = True

[sqlfluff:rules:capitalisation.types]
# Data Types must be uppercase
extended_capitalisation_policy = upper

[sqlfluff:rules:convention.quoted_literals]
force_enable = True
preferred_quoted_literal_style = single_quotes

[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = shorthand

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by moreaupascal56** [comment]
Hi, I have tested this qnd it works we even hqve qn exqmple of materialized view here https://github.com/sqlfluff/sqlfluff/blob/94a1cf701a60ce3bf33f5ec347ce8b484704a62e/test/fixtures/dialects/snowflake/create_view.sql#L82

The example in this issue fails because there is a backtick ` at the end of the query

I think we can close this issue. @WittierDinosaur @arnoprenveille

**Comment by keraion** [comment]
Fixed by #5637
